### PR TITLE
Added instructions for installing and running InstructLAB using WSL

### LINKED ISSUE #1757: Documentation for running on Windows Subsystem for Linux (WSL)
**Is your feature request related to a problem? Please describe.**

Hello, while InstructLab is not supported in Windows, I managed to run it with GPU acceleration (NVIDIA) using an Arch Linux WSL distribution.

Is there an interest in creating a setup guide for this?

If so, I could work on that. I'd just need some guidance on where this would fit in the project (somewhere in the existing md files, or a brand new one?).

### PR DESCRIPTION
**Pull request by rushil-x-gupta**
Added instructions for installing and running InstructLAB using WSL

**Issue resolved by this Pull Request:**
Resolves #1757 

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
Thank you for your contribution! Please make the following changes so the Maintainers can review your PR:
- Squash to a single signed-off commit (`git commit -s`)
- Please move the new file to the existing `docs/` directory

**Comment by rushil-x-gupta** [comment]
Pushed those changes just now! Thank you for taking a look at it, do let me know if there's anything else I should change!

**Comment by jjasghar** [comment]
We should make this very clear this is a non-supported process. 

Maybe this would be better in the Community repo? I really worry about people assuming this is "official. "

File: docs/install-ilab-with-wsl.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+For the sake of simplicity, we will be installing InstructLab using PyTorch without CUDA bindings or GPU acceleration. To do that, first go ahead and install the `python3.10-venv` package.
+
+```
+sudo apt install python3.10-venv
+```
+
+Then,
+
+```
```
Comment (bjhargrave) [inline docs/install-ilab-with-wsl.md:90]: This just looks like a linux without CUAD (CPU only) install. Why not just point to https://github.com/instructlab/instructlab/blob/main/README.md#install-using-pytorch-without-cuda-bindings-and-no-gpu-acceleration instead of repeating the directions here (which are now out of date)?

**Review by bjhargrave — CHANGES_REQUESTED** [review]
I think all you need to do here is to show how to get a WSL environment ready to install instructlab and then point to the existing instructions to install instructlab.

File: docs/install-ilab-with-wsl.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
+# Installing InstructLab on Windows using WSL (Windows Subsystem for Linux)
+
+In this tutorial, I will be walking through the full process of running InstructLab on WSL, from installation of WSL to the initialization of InstructLab. My laptop specs are:
```
Comment (danmcp) [inline docs/install-ilab-with-wsl.md:3]: Avoid first person

File: docs/install-ilab-with-wsl.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+- CPU: Intel i7-10750H
+
+- OS: Windows 10 Home
+
+- RAM: 16GB
+
+- GPU: Geforce RTX 3060 Laptop GPU (12GB VRAM)
+
+Throughout this tutorial, we will be using the following software tools and packages:
```
Comment (danmcp) [inline docs/install-ilab-with-wsl.md:13]: This tutorial will use the following...

File: docs/install-ilab-with-wsl.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+- `cmake`
+
+- `build-essential`
+
+
+## Installing WSL
+
+To install WSL, first open up Powershell. The following command installs the necessary features for WSL and the Ubuntu distro as default. The default distro can be changed with `wsl --list -d <DistributionName>`.
+>NOTE: As of writing, Fedora is not supported with WSL (I could not figure out how to obtain the tar file for it) so I just proceeded with using Ubuntu.
```
Comment (danmcp) [inline docs/install-ilab-with-wsl.md:31]: Avoid first person

File: docs/install-ilab-with-wsl.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+```
+wsl --install
+```
+
+WSL is installed! (Quite easy, wasn’t it?).  If you would still like to install WSL with Fedora, some links that may help are listed below:
```
Comment (danmcp) [inline docs/install-ilab-with-wsl.md:37]: I would remove this block and focus on the paths that are currently supported.

File: docs/install-ilab-with-wsl.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+To run WSL, simply type the following command to set up the Linux environment in Powershell:
+
+```
+wsl
+```
+
+The expected output of this command should be similar to the following:
+
+```
```
Comment (danmcp) [inline docs/install-ilab-with-wsl.md:49]: This block is probably too verbose.  Would remove.

File: docs/install-ilab-with-wsl.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+   just raised the bar for easy, resilient and secure K8s cluster deployment.
+
+   https://ubuntu.com/engage/secure-kubernetes-at-the-edge
+
+This message is shown once a day. To disable it please create the
+/home/user/.hushlogin file.
+```
+
+## Installing InstructLab 
+From here, we can proceed with setting up InstructLab within our Linux environment. The following instructions are a mix of both the official InstructLab documentation as well as the WSL installation/setup process I went through.
```
Comment (danmcp) [inline docs/install-ilab-with-wsl.md:66]: Avoid first person

File: docs/install-ilab-with-wsl.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+```	
+mkdir instructlab
+cd instructlab
+```
+
+Next, update and upgrade the Linux environment to ensure all of your installed packages are up-to-date.
+
+```
+sudo apt update
```
Comment (danmcp) [inline docs/install-ilab-with-wsl.md:78]: These may be good practice but if they aren't required, would avoid general maintenance suggestions

File: docs/install-ilab-with-wsl.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+sudo apt upgrade
+```
+
+For the sake of simplicity, we will be installing InstructLab using PyTorch without CUDA bindings or GPU acceleration. To do that, first go ahead and install the `python3.10-venv` package.
```
Comment (danmcp) [inline docs/install-ilab-with-wsl.md:82]: Avoid we

File: docs/install-ilab-with-wsl.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+
+Then,
+
+```
+python3 -m venv --upgrade-deps venv
+source venv/bin/activate
+pip cache remove llama_cpp_python
+```
+
+Install `cmake`:
```
Comment (danmcp) [inline docs/install-ilab-with-wsl.md:96]: Add these as comments in the script maybe so you can just copy paste all of these scripts in one go

File: docs/install-ilab-with-wsl.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+
+## Running InstructLab
+At last, we can run `instructlab`. Verify the `ilab` CLI is running, then initialize it.
+
+```
+ilab
+ilab config init
+```
+
+And there you go! You have InstructLab setup on your Windows machine (using Linux) through WSL! But it doesn’t stop here. You still need to download your model, generate your synthetic test data, train your model, test it, and even talk to it. If you end up not having enough VRAM to train your models locally (like me), you can use a cloud service like Google Colab. I highly recommend checking out the [official documentation](https://github.com/instructlab/instructlab/tree/main) for these next steps. I will also be releasing more tutorials as I go through my InstructLab journey, so stay tuned!
```
Comment (danmcp) [inline docs/install-ilab-with-wsl.md:122]: Avoid first person

**Review by danmcp — CHANGES_REQUESTED** [review]
Thanks for the submission!

**Comment by nathan-weinberg** [comment]
> Pushed those changes just now! Thank you for taking a look at it, do let me know if there's anything else I should change!

Looks like the commits were not squashed - here are some references that may help you:
- https://cbea.ms/git-commit/
- https://www.git-tower.com/learn/git/faq/git-squash
- https://www.freecodecamp.org/news/git-squash-commits/

Make sure your single squashed commit is signed off according to DCO standards: https://wiki.linuxfoundation.org/dco

**Review by danmcp — CHANGES_REQUESTED** [review]

**Comment by rushil-x-gupta** [comment]
> > Pushed those changes just now! Thank you for taking a look at it, do let me know if there's anything else I should change!
> 
> Looks like the commits were not squashed - here are some references that may help you:
> 
> * https://cbea.ms/git-commit/
> * https://www.git-tower.com/learn/git/faq/git-squash
> * https://www.freecodecamp.org/news/git-squash-commits/
> 
> Make sure your single squashed commit is signed off according to DCO standards: https://wiki.linuxfoundation.org/dco

I'm trying to squash the commits but from my understanding, squashing will squash "up" to the first commit I made, leaving the first commit message as the squashed commit message instead of my latest commit message. I understand that I can reword it to my DCO sign-off when I am using the interactive rebase, but my sign-off will go over the character limit for commit messages.

Is there some sort of workaround for this?

File: docs/install-ilab-with-wsl.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
+# Installing InstructLab on Windows using WSL (Windows Subsystem for Linux)
+
```
Comment (jjasghar) [inline docs/install-ilab-with-wsl.md:2]: Can we put a big "NOT SUPPORTED" or something on this document? Opening up any Windows support should be something we discuss before putting this in any official documentation.

**Comment by nathan-weinberg** [comment]
> > > Pushed those changes just now! Thank you for taking a look at it, do let me know if there's anything else I should change!
> > 
> > 
> > Looks like the commits were not squashed - here are some references that may help you:
> > 
> > * https://cbea.ms/git-commit/
> > * https://www.git-tower.com/learn/git/faq/git-squash
> > * https://www.freecodecamp.org/news/git-squash-commits/
> > 
> > Make sure your single squashed commit is signed off according to DCO standards: https://wiki.linuxfoundation.org/dco
> 
> I'm trying to squash the commits but from my understanding, squashing will squash "up" to the first commit I made, leaving the first commit message as the squashed commit message instead of my latest commit message. I understand that I can reword it to my DCO sign-off when I am using the interactive rebase, but my sign-off will go over the character limit for commit messages.
> 
> Is there some sort of workaround for this?

```bash
$ git reset --soft HEAD~<number of commits you've done>
$ git commit -s
$ git push -f origin <your-branch>
```
